exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/UnexpectedTransactionExecutionException.java

```diff
@@ -41,7 +41,9 @@ public UnexpectedTransactionExecutionException(Throwable cause) {
 
   private static Throwable checkValidCause(Throwable cause) {
     checkNotNull(cause, "null cause is not allowed");
-    checkArgument(!(cause instanceof TransactionExecutionException));
+    // FIXME: revert this change
+    //checkArgument(!(cause instanceof TransactionExecutionException));
+    checkArgument(!(cause instanceof TransactionExecutionException), cause.getClass().toString());
```

Comment:
UTEE test fails with IllegalArgumentException thrown at this line. I added class name to debug it, and it prints `"class com.exonum.binding.core.transaction.TransactionExecutionException"`.

@dmitry-timofeev @bullet-tooth any ideas on how is it possible?

Reply:
🤔 `!` looks redundant here, isn't it?

Reply:
Nevermind, now I understand the problem 😄